app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/onboarding/audit/BusinessPartnerRecordNameMatchDetails.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.models.onboarding.audit
+
+import julienrf.json.derived
+import play.api.libs.json.OFormat
+
+sealed trait BusinessPartnerRecordNameMatchDetails extends Product with Serializable
+
+object BusinessPartnerRecordNameMatchDetails {
+
+  final case class IndividualNameWithSaUtrAuditDetails(firstName: String, lastName: String, sautr: String)
+      extends BusinessPartnerRecordNameMatchDetails
+
+  final case class TrustNameWithTrnAuditDetails(trustName: String, trn: String)
+      extends BusinessPartnerRecordNameMatchDetails
+
+  @SuppressWarnings(Array("org.wartremover.warts.PublicInference"))
+  implicit val format: OFormat[BusinessPartnerRecordNameMatchDetails] =
+    derived.oformat[BusinessPartnerRecordNameMatchDetails]
```

Comment:
You may not want to used the derived format here - it will nest the JSON which we don't want. This: 
```scala
sealed trait A
case class B(i: Int) extends A
implicit val f = derived.oformat[A]
Json.toJson(B(1))
```
will result in this JSON object:
```json
{ "B" : { "i" : 1 } }
```
instead of
```json
{ "i" : 1 }
```

Reply:
It's a sum type so would have to use derived format here unless there is another way?